docs/intro/overrides.rst

```diff
@@ -0,0 +1,349 @@
+allows you to conveniently drill down to which :class:`~.OverrideRule` you're interested in
+using.
+
+.. _`overrides-custom-registry`:
+
+After gathering all the pre-selected rules, we can then store it in a new instance
+of :class:`~.PageObjectRegistry` in order to separate it from the ``default_registry``
+which contains all of the rules. We can use the :meth:`~.PageObjectRegistry.from_override_rules`
+for this:
+
+.. code-block:: python
+
+    from web_poet import PageObjectRegistry
+
+    my_new_registry = PageObjectRegistry.from_override_rules(rules)
+
+
+Handling conflicts from using Multiple External Packages
+--------------------------------------------------------
+
+You might've observed from the previous section that retrieving the list of all
+:class:`~.OverrideRule` from two different external packages may result in a
+conflict. 
+
+We can take a look at the rules for **#2** and **#3** when we were loading all
+available rules:
+
+.. code-block:: python
+
+    # 2. OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'ecommerce_page_objects.site_2.EcomSite2'>, instead_of=<class 'ecommerce_page_objects.EcomGenericPage'>, meta={})
+    # 3. OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'gadget_sites_page_objects.site_2.GadgetSite2'>, instead_of=<class 'gadget_sites_page_objects.GadgetGenericPage'>, meta={})
+
+However, it's technically **NOT** a `conflict`, **yet**, since:
+
+    - ``ecommerce_page_objects.site_2.EcomSite2`` would only be used in **site_2.com**
+      if ``ecommerce_page_objects.EcomGenericPage`` is to be replaced.
+    - The same case with ``gadget_sites_page_objects.site_2.GadgetSite2`` wherein
+      it's only going to be utilized for **site_2.com** if the following is to be
+      replaced: ``gadget_sites_page_objects.GadgetGenericPage``.
+
+It would be only become a conflict if both rules for **site_2.com** `intend to
+replace the` **same** `Page Object`.
+
+However, let's suppose that there are some :class:`~.OverrideRule` which actually
+result in a conflict. To give an example, let's suppose that rules **#2** and **#3**
+`intends to replace the` **same** `Page Object`. It would look something like:
+
+.. code-block:: python
+
+    # 2. OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'ecommerce_page_objects.site_2.EcomSite2'>, instead_of=<class 'common_items.ProductGenericPage'>, meta={})
+    # 3. OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'gadget_sites_page_objects.site_2.GadgetSite2'>, instead_of=<class 'common_items.ProductGenericPage'>, meta={})
+
+Notice that the ``instead_of`` param are the same and only the ``use`` param
+remained different.
+
+There are two main ways we recommend in solving this.
+
+**1. Priority Resolution**
```

Comment:
I think the main use case here could be the following:

1. There is a library of page objects
2. In your project you want to have a different version for one of POs available in the library

In this case there is no need to modify any override rules, it's a matter of using handle_urls decorator in your own project, and setting a priority higher than the default.

Reply:
@kmike to clarify, are you suggesting something like the following?

If we have the conflict like:

```python

rules = default_registry.get_overrides()
print(rules)
# OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'ecommerce_page_objects.site_2.EcomSite2'>, instead_of=<class 'common_items.ProductGenericPage'>, meta={})
# OverrideRule(for_patterns=Patterns(include=['site_2.com'], exclude=[], priority=500), use=<class 'gadget_sites_page_objects.site_2.GadgetSite2'>, instead_of=<class 'common_items.ProductGenericPage'>, meta={})
```

It can be fixed in the project as something like:

```python
import handle_urls

handle_urls(ProductGenericPage, overrides=ProductGenericPage, priority=600)(EcomSite2)
```

At the moment, this won't work since the `PageObjectRegistry` stores the PO class as the key, and if a rule is already existing for a given PO, it ignores new rules attempted to be registered. [Reference](https://github.com/scrapinghub/web-poet/pull/27/files#diff-409cb5fe79e7d7d5ede41109573e31bd847a147a966f477de80c99279bcdf479R153-R161).

Do you think it's worth refactoring this from `Dict[<PO class>, OverrideRule]` into `Dict[<PO class>, List[OverrideRule]]` instead?

Though this would affect the other implementations like `search_overrides()` and how it's being read in **scrapy-poet** as well. Nonetheless, we're still quite early in the process and we can still accommodate this change.

Reply:
@kmike that makes sense! Thanks for explaining it as I initially had the wrong idea. This use case is indeed a common one and it deserves it's own section which I've added in https://github.com/scrapinghub/web-poet/pull/27/commits/7944956802e8c38d939c42b7866d2915f853b1fe.